admin/module/catalog-admin-module-sources/src/main/webapp/js/view/Source.view.js

```diff
@@ -78,6 +78,9 @@ function (ich,Marionette,_,$,Q,ModalSource,EmptyView,Service,Status,wreqr,Utils,
             }
             data.available = this.model.get('available');
             data.name = this.model.get('name');
+            if (!data.name && data.currentConfiguration && data.currentConfiguration.id) {
```

Comment:
Any possibility of cleaning up the entire `serializeData` method?  It looks overcomplicated, and some of the logic just doesn't make sense.  

For instance, there is a check for whether `this.model` exists, but later on it is assumed to.